pages/add/add.jsx

```diff
@@ -1,12 +1,161 @@
+  },
+  handleSignInBtnClick(event) {
+    event.preventDefault();
+
+    let redirectUrl = `${env.ORIGIN}${this.props.router.getCurrentLocation().pathname}`;
+    user.login(redirectUrl);
+  },
+  handleLogOutBtnClick(event) {
+    event.preventDefault();
+
+    user.logout();
+    browserHistory.push({
+      pathname: `/featured`
+    });
+  },
+  handleCreatorClick(event) {
+    event.preventDefault();
+    this.setState({numCreatorFields: this.state.numCreatorFields+1});
+  },
   render() {
+    let creatorFields = [];
+
+    for (var i=0; i<this.state.numCreatorFields; i++) {
+      creatorFields.push( <CreatorInputField key={i} />);
+    }
+
+    let redirectUrl = `${env.ORIGIN}${this.props.router.getCurrentLocation().pathname}`;
+    let anonymousContent = (<HintMessage imgSrc={`/assets/svg/icon-user.svg`}
+                                         header={`Please sign in to add a post`}
+                                         externalLink={user.getLoginURL(redirectUrl)}
+                                         linkText={`Sign in`}
+                                         onClick={this.handleSignInBtnClick}>
+                            <p>Please note, only Mozilla staff can login now as we test this new platform.</p>
+                          </HintMessage>);
+
+
+    let failurePrompt= ( <HintMessage imgSrc={`/assets/svg/icon-user.svg`}
+                                      header={`Sign in failed`}
+                                      internalLink={`/featured`}
+                                      linkText={`Explore featured`}>
+                          <p>Only Mozilla staff can login now as we test this new platform. Check back soon!</p>
+                        </HintMessage>);
+
+    let contentForLoggedInUser = (
+      <div>
+        <h1>Share with the Network</h1>
+        <p>Do you have something to share? If it might be useful to someone in our network, share it here! Pulse includes links to products and software tools, research reports and findings, think pieces, white papers, interviews, and curricula. If it might be useful, share it … at any stage or fidelity.</p>
+        <form action="" method="post">
+          <h2>Basic Info</h2>
+          <div className="posted-by">
+            <p>Posted by:&nbsp;<span className="user-full-name">{user.username}</span></p>
+            <p className="log-out-prompt">Not you? <button className="btn btn-link" onClick={this.handleLogOutBtnClick}>Sign out</button>.</p>
+          </div>
+          <fieldset hidden>
+            <label className="form-control-label">
+              Posted by:&nbsp;
+              <input name="" type="text" className="form-control" value="" readOnly />
+            </label>
+          </fieldset>
+          <label className="form-control-label">
+            Title of the project<em className="required">(required)</em>
+            <input name="title" type="text" placeholder="Title" className="form-control" />
```

Comment:
This is still missing a `required="required"` attribute, to make sure title *must* be filled in.